src/plugins/discover/public/application/apps/main/components/sidebar/lib/group_fields.tsx

```diff
@@ -54,7 +55,8 @@ export function groupFields(
     if (!isFieldFiltered(field, fieldFilterState, fieldCounts)) {
       continue;
     }
-    const isSubfield = useNewFieldsApi && field.spec?.subType?.multi?.parent;
+    const subTypeMulti = getDataViewFieldSubtypeMulti(field?.spec);
+    const isSubfield = useNewFieldsApi && subTypeMulti;
```

Comment:
this is originally checking for` .multi.parent`, the check for parent is missing here

Reply:
Thats happening here - https://github.com/elastic/kibana/pull/112499/files#diff-7cc4a0f9c0c69ffac24b3b3a2956b010c28881bb51b316cd2e3bbe87ef0cb2ebR31